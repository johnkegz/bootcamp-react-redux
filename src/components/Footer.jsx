import React from "react";

export default () => (
  <div>
    <footer className="page-footer relative-bottom  unique-color-dark">
     
      <div className="container">
        <div className="row">
         
          <div className="col-md-12 py-5">
            <div className="mb-5 flex-center">
              
              <a className="fb-ic">
                <i className="fa fa-facebook fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              
              <a className="tw-ic">
                <i className="fa fa-twitter fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              
              <a className="gplus-ic">
                <i className="fa fa-google-plus fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
            </div>
          </div>
          
        </div>
        
      </div>
    <div className="footer-copyright text-center py-3">
      Â© 2018 Copyright:
        <a href="#">stackoverflowlite.com</a>
      </div>
      
    </footer>
  </div>
);
