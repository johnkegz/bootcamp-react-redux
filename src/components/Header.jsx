import React from "react";
import { NavLink } from "react-router-dom";


const Header = () => (
  <header>
    <nav className="navbar fixed-top navbar-expand-lg navbar-dark unique-color-dark scrolling-navbar">
      <div className="container">
        <NavLink to="/home" className="navbar-brand">
          <strong>stackoverflowlite</strong>
        </NavLink>
        <NavLink to="/postQuestion" className="navbar-brand">
          <strong>post question</strong>
        </NavLink>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li>
              <NavLink to={"/login"} className={"navbar-brand"}>
                Login
              </NavLink>
            </li>
            <li>
              <NavLink to={"/signup"} className={"navbar-brand"}>
                signup
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
);
export default Header;
