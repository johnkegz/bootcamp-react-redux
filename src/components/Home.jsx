
import React from "react";
import { ViewAllQuestionsAction } from '../actions/viewAllQuestionsAction';
import { connect } from 'react-redux';
import Moment from "react-moment";
import { Link } from "react-router-dom";
export class Home extends React.Component{
componentDidMount(){     
this.props.ViewAllQuestionsAction();
}
displaySIngleQuestion(questionId){

  window.location.replace(`/singlequestion/${questionId}`);
}
editDataHeading() {
  return (
    <div className="card-header unique-color-dark  white-text text-center mt-0 py-4">
      <img src={"https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"} className="img-responsive" />
    </div>
  );
}

showEditData() {
  
  if(this.props.ViewAllQuestionResults.item){
    
  return Array.from(this.props.ViewAllQuestionResults.item).map(data => (
    <div key={data.question_id}>
   <div className="question">{data.question_id}<span>.</span><Link to={""} onClick={() => this.displaySIngleQuestion(data.question_id)}>
   {data.questions}
              </Link></div>
   <div className="row question-info">
    <div className="row-md-2 date-author">Author:</div>
    <div className="row-md-2 date-author">{data.user_name}</div>
    <div className="row-md-2 date-author">Date:</div>
    <div className="row-md-2 date-author"><Moment format="YYYY/MM/DD">{data.question_date}</Moment></div>
   </div>
   <hr />
   </div>
  ));}
}

render() {

  return (
    <div className="body-container ">
      <div className="row">
        <div className="col-md-12">
          <div className="card cloudy-knoxville-gradient mb-4 wow fadeIn">
            <div>{this.editDataHeading()}</div>
            <div className="card-body">
              <div>{this.showEditData()}</div>
              </div>
          </div>
        </div>
      </div>
    </div>
  );
}
}
export const mapStateToProps = state => {
    return {
      ViewAllQuestionResults: state.allQuestions
    };
  };
export const mapDispatchToProps = dispatch => ({
    ViewAllQuestionsAction: () => dispatch(ViewAllQuestionsAction())
  });
export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(Home);

