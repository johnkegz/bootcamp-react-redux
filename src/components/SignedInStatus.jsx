import React, { Component } from "react";
import { NavLink } from "react-router-dom";



export class SignedInStatus extends Component {
signedIn(){
return <div><ul className="navbar-nav ml-auto">
<li>
<NavLink to="/postQuestion" className="navbar-brand">
          post question
        </NavLink>
</li>

<li>
  <NavLink to={"/signout"} className={"navbar-brand"}>
    sign out
  </NavLink>
</li>
</ul></div>;
}
signedOut(){
  return <div><ul className="navbar-nav ml-auto">
  
  <li>
    <NavLink to={"/login"} className={"navbar-brand"}>
      Login
    </NavLink>
  </li>
  <li>
    <NavLink to={"/signup"} className={"navbar-brand"}>
      signup
    </NavLink>
  </li>
  </ul></div>;
  }
  render() {
    const token = localStorage.getItem("token");

    if (token) {
      return <div>{this.signedIn()}</div>;
    } else {
      return <div>{this.signedOut()}</div>;
    }
  }
}

export default (SignedInStatus);
