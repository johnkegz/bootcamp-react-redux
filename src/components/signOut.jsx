import React from "react";

class SignOut extends React.Component {
  logout(){
    localStorage.clear();
    window.location.href = '/';
  }
  render() {
    return this.logout();
  }
}

export default SignOut;
