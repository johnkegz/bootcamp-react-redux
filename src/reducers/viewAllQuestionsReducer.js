import { ALL_QUESTIONS } from "../actions/types";
  const initialState = {
    item: {},
    
  };
  
  export default function(state = initialState, action) {
    switch (action.type) {
      case ALL_QUESTIONS:
        return {
          ...state,
          item: action.payload
        };
      
      default:
        return state;
    }
  }
  