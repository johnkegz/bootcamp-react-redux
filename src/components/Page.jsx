
import React from "react";
import PropTypes from "prop-types";

import Header from "./Header";
import Footer from "./Footer";

const Page = (props) =>{
  const {children} = props;
  return (
    <div>
      <Header />
      <div className="container">
        {children}
      </div>
      <Footer />
    </div>
  );
};

Page.defaultProps = {
  children: null
};
Page.propTypes = {
  children: PropTypes.element
};
export default Page;
