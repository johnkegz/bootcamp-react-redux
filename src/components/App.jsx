import React from "react";
import { Route, BrowserRouter as Router } from "react-router-dom";
import store from "../store";
import { Provider } from "react-redux";
import Home from "./Home";
import Page from "./Page";
import SignUp from "../containers/signUp/SignUp";
import Login from "../containers/login/Login";

import postQuestion from "../containers/questions/postQuestion";
const App = () => (
  <Provider store={store}>
    <Router>
      <React.Fragment>
        <Page>
          <div>
            <Route exact path="/" component={Home} />
            <Route path="/home" component={Home} />
            <Route path="/signup" component={SignUp} />
            <Route path="/login" component={Login} />
            <Route path="/postQuestion" component={postQuestion} />
          </div>
        </Page>
      </React.Fragment>
    </Router>
  </Provider>
);

export default App;
