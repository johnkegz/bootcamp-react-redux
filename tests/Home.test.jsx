import React from "react";
import { shallow } from "enzyme";
import { Home } from "../src/components/Home";
import viewAllQuestionsReducer from "../src/reducers/viewAllQuestionsReducer";

it("login test", () => {
  const props = {
    
    ViewAllQuestionResults: {item:{question_date: "Tue, 02 Oct 2018 00:00:00 GMT",
    question_id: 1,
    questions: "what is andela",
    user_id: 1,
    user_name: "kegz"}},
    ViewAllQuestionsAction: data => jest.fn()
  };

  const component = shallow(<Home {...props} />);
  expect(component).toMatchSnapshot();
});

it("Test ViewAllQuestions reducder", () => {
  const initialState = {
    item: {}
  };
  const action = {
    type: "ALL_QUESTIONS",
    payload: {}
  };
  expect(viewAllQuestionsReducer(undefined, action)) !== initialState;
});

