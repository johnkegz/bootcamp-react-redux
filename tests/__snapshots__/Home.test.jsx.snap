// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    ViewAllQuestionResults={
      Object {
        "item": Object {
          "question_date": "Tue, 02 Oct 2018 00:00:00 GMT",
          "question_id": 1,
          "questions": "what is andela",
          "user_id": 1,
          "user_name": "kegz",
        },
      }
    }
    ViewAllQuestionsAction={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
          >
            <div>
              <div
                className="card-header unique-color-dark  white-text text-center mt-0 py-4"
              >
                <img
                  className="img-responsive"
                  src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                />
              </div>
            </div>
            <div
              className="card-body"
            >
              <div />
            </div>
          </div>
        </div>
      </div>,
      "className": "body-container ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-12"
        >
          <div
            className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
          >
            <div>
              <div
                className="card-header unique-color-dark  white-text text-center mt-0 py-4"
              >
                <img
                  className="img-responsive"
                  src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                />
              </div>
            </div>
            <div
              className="card-body"
            >
              <div />
            </div>
          </div>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
          >
            <div>
              <div
                className="card-header unique-color-dark  white-text text-center mt-0 py-4"
              >
                <img
                  className="img-responsive"
                  src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                />
              </div>
            </div>
            <div
              className="card-body"
            >
              <div />
            </div>
          </div>,
          "className": "col-md-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <div
                  className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                >
                  <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />
                </div>
              </div>,
              <div
                className="card-body"
              >
                <div />
              </div>,
            ],
            "className": "card cloudy-knoxville-gradient mb-4 wow fadeIn",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                >
                  <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />,
                  "className": "card-header unique-color-dark  white-text text-center mt-0 py-4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "img-responsive",
                    "src": "https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div />,
                "className": "card-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <div
              className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
            >
              <div>
                <div
                  className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                >
                  <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />
                </div>
              </div>
              <div
                className="card-body"
              >
                <div />
              </div>
            </div>
          </div>
        </div>,
        "className": "body-container ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-md-12"
          >
            <div
              className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
            >
              <div>
                <div
                  className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                >
                  <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />
                </div>
              </div>
              <div
                className="card-body"
              >
                <div />
              </div>
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card cloudy-knoxville-gradient mb-4 wow fadeIn"
            >
              <div>
                <div
                  className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                >
                  <img
                    className="img-responsive"
                    src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                  />
                </div>
              </div>
              <div
                className="card-body"
              >
                <div />
              </div>
            </div>,
            "className": "col-md-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <div
                    className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                  >
                    <img
                      className="img-responsive"
                      src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                    />
                  </div>
                </div>,
                <div
                  className="card-body"
                >
                  <div />
                </div>,
              ],
              "className": "card cloudy-knoxville-gradient mb-4 wow fadeIn",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card-header unique-color-dark  white-text text-center mt-0 py-4"
                  >
                    <img
                      className="img-responsive"
                      src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                    />
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="img-responsive"
                      src="https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg"
                    />,
                    "className": "card-header unique-color-dark  white-text text-center mt-0 py-4",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "img-responsive",
                      "src": "https://res.cloudinary.com/dksxmwjqs/image/upload/v1546420242/zstzna8xsm5zv8c5laxm.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div />,
                  "className": "card-body",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
